### PR TITLE
yum-cron: reintroduce days_of_week parameter

### PR DESCRIPTION
**Pull request by whereismyjetpack**
In previous versions of yum_cron we had available to us the days_of_week parameter. This patch reintroduces that behavior, and adds the default days_of_week setting in the sample configuration files.

### CODE REVIEW

File: etc/yum-cron-hourly.conf [hunk 1]
```diff
@@ -20,6 +20,9 @@ download_updates = no
 # the update to be applied
 apply_updates = no
 
+# Days of week
```
Comment (dmnks) [inline etc/yum-cron-hourly.conf]: Please provide a better documentation here. You can use the one from the original DAYS_OF_WEEK option:
`You may set DAYS_OF_WEEK to the numeric days of the week you want to run, where 0 is Sunday and 6 is Saturday. The default is to run every day.`

File: etc/yum-cron-security.conf [hunk 2]
```diff
@@ -20,6 +20,9 @@ download_updates = yes
 # the update to be applied
 apply_updates = yes
 
+# Days of week
```
Comment (dmnks) [inline etc/yum-cron-security.conf]: Same as above.

File: etc/yum-cron.conf [hunk 3]
```diff
@@ -19,6 +19,9 @@ download_updates = yes
 # that download_updates must also be yes for the update to be applied.
 apply_updates = no
 
+# Days of week
```
Comment (dmnks) [inline etc/yum-cron.conf]: Same as above.

File: etc/yum-cron-hourly.conf [hunk 4]
```diff
@@ -20,6 +20,9 @@ download_updates = no
...
+days_of_week="0123456"
```
Comment (dmnks) [inline etc/yum-cron-hourly.conf]: Please add spaces before and after the `=` to make it consistent with the other options.

**Review by dmnks — COMMENTED** [review]
Thanks, see my comments inline. Other than that, looks fine.

**Review by dmnks — CHANGES_REQUESTED** [review]
(Actually requesting changes)

**Comment by whereismyjetpack** [comment]
Thanks for the review. I've pushed up the required changes, please let me know if you need anything else.

File: etc/yum-cron-hourly.conf [hunk 5]
```diff
@@ -20,8 +20,8 @@ download_updates = no
 # the update to be applied
 apply_updates = no
 
-# Days of week
-days_of_week="0123456"
+# You may set days_of_week to the numeric days of the week you want to run, where 0 is Sunday and 6 is Saturday. The default is to run every day.
```
Comment (dmnks) [inline etc/yum-cron-hourly.conf]: Awsome, one minor nitpick, though: please wrap the line to 80 chars :) (again, for consistency and ease of reading)

**Review by dmnks — CHANGES_REQUESTED** [review]
Much better, please see the nitpick and, please, squash this commit onto the previous one (use git push --force).

**Comment by whereismyjetpack** [comment]
Should be all fixed up now. Thanks!

File: etc/yum-cron-hourly.conf [hunk 6]
```diff
@@ -20,6 +20,10 @@ download_updates = no
 # the update to be applied
 apply_updates = no
 
+# You may set days_of_week to the numeric days of the week you want to run,
+# where 0 is Sunday and 6 is Saturday. The default is to run every day.
```
Comment (dmnks) [inline etc/yum-cron-hourly.conf]: date.weekday() starts on Monday so Sunday is 6 (not 0), see:
https://docs.python.org/2/library/datetime.html#datetime.date.weekday

**Review by dmnks — CHANGES_REQUESTED** [review]
One more issue, please see inline.

File: etc/yum-cron-hourly.conf (hunk 6 above)
Comment (whereismyjetpack) [reply etc/yum-cron-hourly.conf]: Sorry for the delay. I'm not sure how i'd like to handle this. Should we break backwards compatibility with yum-cron that ships with el6, or refactor so 0 is Sunday. 

I'm leaning on updating the docs, and breaking the precedent. What do you all think?

**Comment by whereismyjetpack** [comment]
I've committed the changes to docs. Thanks!

**Comment by dmnks** [comment]
Actually, you make a good point regarding the el6 version, which I didn't realize initially. The earlier bash version of yum-cron in el6 used `date +%w` which indeed returns 0 for Sunday. After all, in `crontab`, day of week also has 0 for Sunday.

That said, I would rather lean towards making `yum-cron` compatible with both the el6 version and `crontab`. Could you maybe update the patch accordingly? One way is replacing `weekday()` with `isoweekday() % 7` which should give you the original range 0..6 where 0 is Sunday.

Thank you again!

**Comment by coreypenford** [comment]
@whereismyjetpack Is there a possibility of this getting picked up again and merged? If not I may open my own PR with your changes + @dmnks's requested ones. Pretty basic functionality that would be nice to have

**Comment by dmnks** [comment]
Thank you for the PR. However, please note that this project has been deprecated and is no longer maintained upstream. Please consider migrating to [DNF](https://github.com/rpm-software-management/dnf) which is the successor of YUM. That said, I’ll close this PR now.

**Pull request closed without merging**